DNN Platform/Library/Entities/Portals/StyleColor.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        public StyleColor()
+        {
+            this.hex = "FFFFFF";
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StyleColor"/> struct.
+        /// </summary>
+        /// <param name="hexValue">The hex value to use.</param>
+        public StyleColor(string hexValue)
+        {
+            if (this.IsValidCssColor(hexValue))
+            {
+                this.HexValue = ExpandColor(hexValue);
+            }
+            else
+            {
+                this.HexValue = ExpandColor("FFFFFF");
+            }
+        }
+
+        private enum Component
+        {
+            Red,
+            Green,
+            Blue,
+        }
+
+        /// <inheritdoc/>
+        public byte Red
+        {
+            get { return this.GetComponent(Component.Red); }
+            set { this.SetComponent(Component.Red, value); }
+        }
+
+        /// <inheritdoc/>
+        public byte Green
+        {
+            get { return this.GetComponent(Component.Green); }
+            set { this.SetComponent(Component.Green, value); }
+        }
+
+        /// <inheritdoc/>
+        public byte Blue
+        {
+            get { return this.GetComponent(Component.Blue); }
+            set { this.SetComponent(Component.Blue, value); }
+        }
+
+        /// <inheritdoc/>
+        public string HexValue
+        {
+            get
+            {
+                return this.hex;
+            }
+
+            set
+            {
+                if (this.IsValidCssColor(value))
+                {
+                    this.hex = ExpandColor(value);
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public string MinifiedHex
+        {
+            get
+            {
+                if (
+                    this.hex[0] == this.hex[1] &&
+                    this.hex[2] == this.hex[3] &&
+                    this.hex[4] == this.hex[5])
+                {
+                    return this.hex[0].ToString() + this.hex[2].ToString() + this.hex[4].ToString();
+                }
+
+                return this.hex;
+            }
+        }
+
+        private static string ExpandColor(string hexValue)
+        {
+            if (hexValue.Length == 6)
+            {
+                return hexValue.ToUpperInvariant();
+            }
+
+            string value;
+            var r = hexValue[0];
+            var g = hexValue[1];
+            var b = hexValue[2];
+            value = string.Concat(r, r, g, g, b, b);
+            return value.ToUpperInvariant();
+        }
+
+        private bool IsValidCssColor(string hexValue)
+        {
+            if (string.IsNullOrWhiteSpace(hexValue))
+            {
+                throw new ArgumentNullException("You need to provide a CSS color value in the constructor");
+            }
+
+            var regex = new Regex(@"([\da-f]{3}){1,2}", RegexOptions.IgnoreCase);

```

Comment:
Should this Regex instance be made a static field (maybe with RegexOptions.Compiled) instead of recreating it every time that IsValidCssColor is called?

Reply:
Good idea, done in https://github.com/dnnsoftware/Dnn.Platform/pull/5851/commits/0a675fae3646d49e8100f225414fa35e227c401e